### ISSUE TITLE
cannot assign to `f`: expected `Foo`, not `any_int`

### ISSUE THREAD

**Issue by danieldaeschle** [vlang/v#6748]
**V version:** custom
**OS:** Windows 10 WSL2 Ubuntu 20.04

**What did you do?**
```v
type Foo = int | string

fn main() {
	mut f := Foo(5)
	f = 6
}
```

**What did you expect to see?**
that f has the value 6
 
**What did you see instead?**
```
main.v:5:6: error: cannot assign to `f`: expected `Foo`, not `any_int`
    3 | fn main() {
    4 |     mut f := Foo(5)
    5 |     f = 6
      |         ^
    6 | }
```

**Comment by StunxFS** [comment]
With this if it works:
```v
type Foo = int | string

fn main() {
	mut f := Foo(5)
	f = Foo(6)
}
```

**Comment by danieldaeschle** [comment]
i know

**Comment by ntrel** [comment]
`any_int` strikes again. The parser could parse integer literals as `int` unless the value exceeds `math.max_i32`.

**Comment by spaceface777** [comment]
Closed in favor of #6575.
